File: typescript/rest-api/functions/read.ts
// [START snippet]
// functions/read.ts

import { faas, documents } from "@nitric/sdk";
import { path } from "../common";

// Start your function here
faas
  .http(async (ctx: faas.HttpContext): Promise<faas.HttpContext> => {
    // get params from path
    const { orderId } = path.test(ctx.req.path);

    if (!orderId) {
      ctx.res.body = new TextEncoder().encode("Invalid request");
      ctx.res.status = 400;
    }

    const orders = documents().collection("orders");

    try {
      const order = await orders.doc(orderId).get();
      ctx.res.json(order);
    } catch {
      ctx.res.status = 404;
      ctx.res.json({ message: "order not found" });
    }

    return ctx;
  })
  .start();

// [END snippet]  

File: python/rest-api/functions/read.py
# [START snippet]

import json

from nitric.faas import HttpContext, http
from nitric.api import Documents

async def httpHandler(ctx: HttpContext) -> HttpContext:
  order_id = ctx.req.path.split("/")[-1]  # use a parser in a real project.
  orders = Documents().collection("orders")

  try:
    order = await orders.doc(order_id).get()

    ctx.res.status = 200
    ctx.res.body = json.dumps(order.content).encode('utf-8')
  except:
    ctx.res.status = 404
    ctx.res.body = 'Order with id: {order_id} not found'.encode("utf-8")
  
  return ctx

if __name__ == "__main__":
  http(httpHandler).start()

# [END snippet]